crates/bevy_transform/src/hierarchy/hierarchy_maintenance_system.rs

```diff
@@ -67,20 +67,96 @@ pub fn parent_update_system(
     // collect multiple new children that point to the same parent into the same
     // SmallVec, and to prevent redundant add+remove operations.
     children_additions.iter().for_each(|(e, v)| {
-        commands.entity(*e).insert(Children::with(v));
+        // Mark the entity with a `DirtyParent` component so that systems which run
+        // in the same stage have a way to query for a missed update.
```

Comment:
I feel like this comment is more appropriate on the struct declaration. Right now the only explanation as to what `DirtyParent` means is here but the struct itself has no documentation.

Reply:
I'm not certain the struct is a great place for this particular bit of documentation - `DirtyParent` is a quirk of both the `parent_update_system` and how the stages are currently run; the comment is there because it's not obvious why you'd add it in that particular code branch.

As to the wider point, I don't know what the documentation should achieve. The problem is quite a niche one - you need to spawn your components in a particular way and even then, you'll only touch `DirtyParent` in the specific case that you're running a system in the same stage. If you're looking at `DirtyParent` directly, it's likely that you'll want to read through the code itself. If not, then arguably it should be part of the documentation on the whole hierarchy system.